### PR TITLE
Added missing lines

### LINKED ISSUE #13217: [GeoMechanicsApplication] Create integration tests for the Mohr-Coulomb model
This issue includes registering the new Mohr-Coulomb constitutive law.

Create the following related tests (using axisymmetric and plane strain elements, 1 quad element per test):
1. Triaxial tension
2. Triaxial compression
3. Shear

Don't forget to document the tests.

# Acceptance criteria

The test results should compare with the equivalent test results obtained with Plaxis. We should use the soil test facility provided by Plaxis.

### PR DESCRIPTION
**Pull request by mnabideltares**
**📝 Description**
Missing lines in Moh-Coulomb are added.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        "nodal_smoothing":                    true,
+        "block_builder":                      true,
+        "solution_type":                      "Quasi-Static",
+        "scheme_type":                        "Newmark",
+        "reset_displacements":                true,
+        "newmark_beta":                       0.25,
+        "newmark_gamma":                      0.5,
+        "newmark_theta":                      0.5,
+        "rayleigh_m":                         0.0,
+        "rayleigh_k":                         0.0,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Not used, as the computation has solution_type Quasi-Static

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        "strategy_type":                      "newton_raphson",
+        "convergence_criterion":              "residual_criterion",
+        "displacement_relative_tolerance":    1.0E-9,
+        "displacement_absolute_tolerance":    1.0E-9,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Not used, as the convergence_criterion is residual_criterion.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        "desired_iterations":                 4,
+        "max_radius_factor":                  10.0,
+        "min_radius_factor":                  0.1,
+        "calculate_reactions":                true,
+        "max_line_search_iterations":         5,
+        "first_alpha_value":                  0.5,
+        "second_alpha_value":                 1.0,
+        "min_alpha":                          0.1,
+        "max_alpha":                          2.0,
+        "line_search_tolerance":              0.5,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Not used, as the strategy_type is newton_raphson.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        "rotation_dofs":                      true,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Why true, there are no elements with rotation D.o.F. in this computation.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        "linear_solver_settings":             {
+            "solver_type":   "LinearSolversApplication.sparse_lu",
+            "scaling":       true
+        },
+        "problem_domain_sub_model_part_list": ["Soil"],
+        "processes_sub_model_part_list":      ["Soil", "Bottom","Sides", "Gravity"],
+        "body_domain_sub_model_part_list":    ["Soil"]
+    },
+    "output_processes": {
+        "gid_output": [{
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json:61]: The test uses get_on_integration_points to retrieve effective stresses. None of the gid_output is used for comparision.

**Review by WPK4FEM — COMMENTED** [review]
Please make the test files as short as possible.

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp [hunk 6]
```diff
@@ -209,12 +225,13 @@ void MohrCoulombWithTensionCutOff::InitializeMaterialResponseCauchy(Parameters&
 
 void MohrCoulombWithTensionCutOff::GetLawFeatures(Features& rFeatures)
 {
-    rFeatures.SetOptions(mpConstitutiveDimension->GetSpatialType());
-    rFeatures.SetOptions(INFINITESIMAL_STRAINS);
-    rFeatures.SetOptions(ISOTROPIC);
+    Flags cl_options;
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp]: Perhaps simply `options` is good enough here? (I generally don't like abbreviations like `cl`.)

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp [hunk 7]
```diff
@@ -209,12 +225,13 @@ void MohrCoulombWithTensionCutOff::InitializeMaterialResponseCauchy(Parameters&
...
+    cl_options.Set(mpConstitutiveDimension->GetSpatialType());
+    cl_options.Set(ConstitutiveLaw::INFINITESIMAL_STRAINS, true);
+    cl_options.Set(ConstitutiveLaw::ISOTROPIC, true);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp]: It appears we can leave out the value `true`, which is simpler and would make it more similar to the statement at line 229.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 8]
```diff
@@ -19,6 +19,7 @@
 #include "tests/cpp_tests/test_utilities.h"
 
 #include <boost/numeric/ublas/assignment.hpp>
+#include <custom_constitutive/three_dimensional.h>
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: For non-system headers, we use double quotes rather than angular brackets:
```suggestion
#include "custom_constitutive/three_dimensional.h"
```

Also, I'd suggest to keep the "local" headers together. In other words, I'd move this line to just below line 19.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 9]
```diff
@@ -388,4 +389,135 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    Properties properties;
+    properties.SetValue(YOUNG_MODULUS, 1.0e8);
+    properties.SetValue(POISSON_RATIO, 0.3);
+    properties.SetValue(GEO_FRICTION_ANGLE, 35.0);
+    properties.SetValue(GEO_COHESION, 10.0);
+    properties.SetValue(GEO_DILATANCY_ANGLE, 20.0);
+    properties.SetValue(GEO_TENSILE_STRENGTH, 10.0);
+    ConstitutiveLaw::Parameters parameters;
+    parameters.SetMaterialProperties(properties);
+    parameters.SetOptions(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Perhaps it's clearer to use the mechanism that _adds_ options rather than _overwriting_ them, i.e.:
```suggestion
    parameters.GetOptions().Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
```

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 10]
```diff
@@ -388,4 +389,135 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    parameters.SetConstitutiveMatrix(constitutive_matrix);
+
+    const auto dummy_element_geometry      = Geometry<Node>{};
+    const auto dummy_shape_function_values = Vector{};
+    law.InitializeMaterial(properties, dummy_element_geometry, dummy_shape_function_values);
+
+    // Act
+    Vector     cauchy_stress_vector = ZeroVector(4);
+    const auto dummy_process_info   = ProcessInfo{};
+    law.SetValue(CAUCHY_STRESS_VECTOR, cauchy_stress_vector, dummy_process_info);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: I'm not sure whether we'd need this function call for this test. Do you know? (It continues to pass when I comment it out.)

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 11]
```diff
@@ -388,4 +389,135 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    KRATOS_EXPECT_MATRIX_NEAR(constitutive_matrix, expected_constitutive_matrix, 1.E6);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_WorkingSpaceDimension, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    auto mc_law = MohrCoulombWithTensionCutOff(std::make_unique<PlaneStrain>());
+
+    // Act & Assert
+    KRATOS_EXPECT_EQ(mc_law.WorkingSpaceDimension(), 2);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Perhaps we'd also like to assert/show when this constitutive law is used in a 3D model, the returned value equals 3?
```suggestion
    KRATOS_EXPECT_EQ((MohrCoulombWithTensionCutOff{std::make_unique<PlaneStrain>()}.WorkingSpaceDimension()), 2);
    KRATOS_EXPECT_EQ((MohrCoulombWithTensionCutOff{std::make_unique<ThreeDimensional>()}.WorkingSpaceDimension()), 3);
```

We could consider to extend this idea to some of the other tests as well.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 12]
```diff
@@ -388,4 +389,135 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    auto mc_law_3d = MohrCoulombWithTensionCutOff(std::make_unique<ThreeDimensional>());
+
+    // Act
+    mc_law_3d.GetLawFeatures(features);
+
+    // Assert
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::ISOTROPIC))
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::THREE_DIMENSIONAL_LAW))
+    KRATOS_EXPECT_EQ(features.GetStrainMeasures().front(),ConstitutiveLaw::StrainMeasure_Infinitesimal);
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: I think we need to rerun `kp format`, because I reckon the formatting of this line does not comply with what Clang-format does.

**Review by avdg81 — COMMENTED** [review]
I'm very pleased to see the additional (unit) tests, which truly helps to build confidence in our code. I'm also see happy to see that the documentation has been improved and that the code has been cleaned up a bit. I have several minor suggestions, none of them being blocking.

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp (hunk 6 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp (hunk 7 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 8 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 9 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 10 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 11 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 12 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json (hunk 1 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Done.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json (hunk 2 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Done.

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp [hunk 13]
```diff
@@ -209,12 +225,13 @@ void MohrCoulombWithTensionCutOff::InitializeMaterialResponseCauchy(Parameters&
 
 void MohrCoulombWithTensionCutOff::GetLawFeatures(Features& rFeatures)
 {
-    rFeatures.SetOptions(mpConstitutiveDimension->GetSpatialType());
-    rFeatures.SetOptions(INFINITESIMAL_STRAINS);
-    rFeatures.SetOptions(ISOTROPIC);
+    auto options = Flags{};
+    options.Set(mpConstitutiveDimension->GetSpatialType());
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp:229]: Just for my understanding: does this mean that before the options were overwritten?

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 14]
```diff
@@ -388,4 +389,121 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    KRATOS_EXPECT_EQ(MohrCoulombWithTensionCutOff{std::make_unique<PlaneStrain>()}.GetStrainMeasure(),
+                     ConstitutiveLaw::StrainMeasure_Infinitesimal);
+    KRATOS_EXPECT_EQ(MohrCoulombWithTensionCutOff{std::make_unique<ThreeDimensional>()}.GetStrainMeasure(),
+                     ConstitutiveLaw::StrainMeasure_Infinitesimal);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_HasIncrementalFormulation, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    EXPECT_TRUE(MohrCoulombWithTensionCutOff{std::make_unique<PlaneStrain>()}.IsIncremental());
+    EXPECT_TRUE(MohrCoulombWithTensionCutOff{std::make_unique<ThreeDimensional>()}.IsIncremental());
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp:467]: Since these are not really dependent on the ConstitutiveDimension, it doesn't add any coverage to check it twice. I'm fine with keeping it, but it might make more sense to only do one check (same holds for the rest of the tests where the results do not depend on the dimension)

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp [hunk 15]
```diff
@@ -388,4 +389,121 @@ KRATOS_TEST_CASE_IN_SUITE(MohrCoulombWithTensionCutOff_Serialization, KratosGeoM
...
+    mc_law_3d.GetLawFeatures(features);
+
+    // Assert
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::ISOTROPIC))
+    KRATOS_EXPECT_TRUE(features.GetOptions().Is(ConstitutiveLaw::THREE_DIMENSIONAL_LAW))
+    KRATOS_EXPECT_EQ(features.GetStrainMeasures().front(), ConstitutiveLaw::StrainMeasure_Infinitesimal);
+    KRATOS_EXPECT_EQ(features.GetStrainSize(), 6);
+    KRATOS_EXPECT_EQ(features.GetSpaceDimension(), 3);
+}
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Personally I'd split this into two tests (then it's more clear when e.g. one of them fails)

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/README.md [hunk 16]
```diff
@@ -23,4 +23,4 @@ The boundary conditions are shown below:
 
 All degrees of freedom have been prescribed. In horizontal and vertical directions, the element is compressed or extended by imposing displacements. The elastic behavior of the elements is described by a Young's modulus of $`1000 \mathrm{Pa}`$ and a Poisson's ratio of $`0.0`$. The material behavior is described using a cohesion of $`10.0 \mathrm{Pa}`$, a friction angle of $`35^{\circ}`$, a dilatancy angle of $`20^{\circ}`$ and a tensile strength of $`10 \mathrm{Pa}`$.
 
-These tests are also extended to three dimensional elements. The 3D cases are applied on hexahydrons of 8 nodes, namely 3D8N.
\ No newline at end of file
+These tests are also extended to three dimensional elements. The 3D cases are applied on 8 node hexahedra, namely 3D8N.
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/README.md]: We could also add a few sentences about the new test (since it's a bit different than the rest)

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+                            "MultiFileFlag":         "SingleFile"
+                        },
+                        "file_label":          "step",
+                        "output_control_type": "step",
+                        "output_interval":    1,
+                        "body_output":         true,
+                        "node_output":         false,
+                        "skin_output":         false,
+                        "nodal_results":       ["DISPLACEMENT","NODAL_CAUCHY_STRESS_TENSOR"],
+                        "gauss_point_results": ["GREEN_LAGRANGE_STRAIN_TENSOR","CAUCHY_STRESS_TENSOR","TOTAL_STRESS_TENSOR"]
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: I think this can still be cleaned up. Maybe the entire process can be removed (depending on how the test retrieves things), but at least we only need to output stresses

**Review by rfaasse — DISMISSED** [review]
Looks good! I have a few minor suggestions, but non-blocking, so feel free to either take them into account, or leave them

File: applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp (hunk 13 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp:229]: Yes, that is correct. @WPK4FEM discovered this while he was writing unit tests.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 14 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp:467]: Indeed, it doesn't add anything to the code coverage, but it does communicate that some member functions return results that depend on the constitutive law dimension that was passed to the constructor, and other member functions return results that are independent of that. From a documentation point of view, I'd like to keep these additional assertions. Also note that the test case names reflect what to expect. What do you think?

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 15 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp]: Done.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json (hunk 17 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: I have tried to remove the entire GiD output process, but in that case the test fails since retrieving the stresses from the database through the test helper requires a GiD output process to be defined to get the model part:
```
======================================================================
ERROR: test_column_under_gravity (__main__.KratosGeoMechanicsMohrCoulombWithTensionTests.test_column_under_gravity)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Checkouts\KratosProjects\rome\bin\Debug\applications\GeoMechanicsApplication\tests\test_mohr_coulomb_with_tension_cutoff.py", line 73, in test_column_under_gravity
    sig_xx, sig_yy = self.simulate_mohr_coulomb('test_column_under_gravity', 2)
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Checkouts\KratosProjects\rome\bin\Debug\applications\GeoMechanicsApplication\tests\test_mohr_coulomb_with_tension_cutoff.py", line 15, in simulate_mohr_coulomb
    cauchy_stresses = test_helper.get_on_integration_points(simulation, Kratos.CAUCHY_STRESS_TENSOR)
  File "C:\Checkouts\KratosProjects\rome\bin\Debug\applications\GeoMechanicsApplication\tests\test_helper.py", line 308, in get_on_integration_points
    model_part = simulation._list_of_output_processes[0].model_part
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

Therefore, I will limit the requested output to `CAUCHY_STRESS_TENSOR`.

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/README.md (hunk 16 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/README.md]: Done.

File: applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp (hunk 14 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_mohr_coulomb_with_tension_cutoff.cpp:467]: For me its fine either way, so let's keep it like this

File: applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json (hunk 17 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_mohr_coulomb_with_tension_cutoff/test_column_under_gravity/ProjectParameters.json]: Yep, we need to change that at some point, but for now it's fine

**Review by rfaasse — APPROVED** [review]
Good to go!
